### ISSUE TITLE
Bugfix - get current request when called internally

### ISSUE THREAD

**Issue by awalko** [chrisbjr/api-guard#2]
Hi,

I started using api-guard to secure my API with keys and limits and it works great !

I just have two issues with it, related to the authorization header:

The first one is that you need to name it x-something if you want the call Request::header(Config::get('api-guard::keyName')); work in ApiGuardController.php line 71 (the default value 'Authorization' does not work).

The second one is when you call your API internally, and try to pass the key as the header of your request, like this:

```
          $request = \Request::create('/my/custom/route/to/my/api/action', 'GET');
          $request->headers->set('x-authorization','80c10c72e64eaafc0a644db203432c1f9');
          $response = \Route::dispatch($request);
```

Then the same line in your ApiGuardController.php should be:

 $key = Route::getCurrentRequest()->header(Config::get('api-guard::keyName'));

otherwise the request checked is not the good one - which the one we pass to the API.

Do you think you could patch the code with this small change ? 

Basically anywhere you call Request::something you should replace it by Route::getCurrentRequest->something to be sure.

Thank you,
